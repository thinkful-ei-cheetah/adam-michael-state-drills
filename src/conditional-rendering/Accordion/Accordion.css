.accordion-section {
  border: 1px solid black;
  padding: 15px;
  background-color: lightblue;
}