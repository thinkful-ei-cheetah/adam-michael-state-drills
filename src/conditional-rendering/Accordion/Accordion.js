import React from 'react';
import './Accordion.css'
import Section from '../Section/Section';

export default class Accordion extends React.Component {
  constructor(props) {
    super(props)
    
    this.state = {
      sections: this.props.sections,
      clickedView: null
    }
  }

  showSection = (value) => {
    this.setState({
      clickedView: value
    })
  }

  render() {
    const accordion = this.state.sections.map((section, index) => {
      return (
        < Section 
          title={section.title}
          content={section.content}
          key={index}
          id={index}
          showSection={this.showSection} 
          clickedView={this.state.clickedView}
        />
      )
    })

    return (
      <ul>
        {accordion.length ? accordion : <li></li>}
      </ul>
    );
  }
}