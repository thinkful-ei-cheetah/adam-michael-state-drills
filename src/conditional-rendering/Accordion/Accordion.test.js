import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

const sections = [
  {title: 'Foo', content: 'Hey, this is the content for Foo'},
  {title: 'Bar', content: 'And, this is the better content for Bar'},
  {title: 'Baz', content: 'Andddd, this is the best content for Baz'}
]

describe('Accordion Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion sections={sections} />, div)
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the correct amount of Section components initially', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJson(wrapper).node.rendered.length).toBe(sections.length)
  })

  describe('when an accordion button is clicked', () => {
    it('sets clickView state to section id', () => {
      const wrapper = shallow(<Accordion sections={sections} />)
      wrapper.find('Section').at(0).dive().find('button').simulate('click');
      expect(toJson(wrapper)).toMatchSnapshot();
    })
  })

  describe('when an accordion button is clicked', () => {
    it('shows the correct content', () => {
      const wrapper = shallow(<Accordion sections={sections} />)
      wrapper.find('Section').at(0).dive().find('button').simulate('click');
      const section = wrapper.find('Section').at(0).dive()
      expect(toJson(section)).toMatchSnapshot();
    })
  })

  describe('when an empty array is given', () => {
    it('renders a single empty li', () => {
      const wrapper = shallow(<Accordion sections={[]}/>)
      expect(toJson(wrapper)).toMatchSnapshot();
    })
  })
})

