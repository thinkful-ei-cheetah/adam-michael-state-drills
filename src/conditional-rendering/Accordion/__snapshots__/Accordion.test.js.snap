// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component when an accordion button is clicked sets clickView state to section id 1`] = `
<ul>
  <Section
    clickedView={0}
    content="Hey, this is the content for Foo"
    id={0}
    key="0"
    showSection={[Function]}
    title="Foo"
  />
  <Section
    clickedView={0}
    content="And, this is the better content for Bar"
    id={1}
    key="1"
    showSection={[Function]}
    title="Bar"
  />
  <Section
    clickedView={0}
    content="Andddd, this is the best content for Baz"
    id={2}
    key="2"
    showSection={[Function]}
    title="Baz"
  />
</ul>
`;

exports[`Accordion Component when an accordion button is clicked shows the correct content 1`] = `
<li
  className="accordion-section"
>
  <button
    onClick={[Function]}
  >
    Foo
  </button>
  <p>
    Hey, this is the content for Foo
  </p>
</li>
`;

exports[`Accordion Component when an empty array is given renders a single empty li 1`] = `
<ul>
  <li />
</ul>
`;
